velox/common/memory/RawVector.h

```diff
@@ -127,12 +139,14 @@ class raw_vector {
 
   void reserve(int32_t size) {
     if (capacity_ < size) {
-      T* newData = allocateData(size, capacity_);
+      T* newData = allocateData(size);
+      const auto newCapacity = calculateCapacity(size);
```

Comment:
this might leads to inconsistent state. Let's don't update object state in allocateData and handle exception in followup execution to free newData. Also let's use calculateCapacity to calculate capacity in all places.